packages/twenty-front/src/modules/settings/admin-panel/components/SettingsAdminGeneral.tsx

```diff
@@ -0,0 +1,268 @@
+import { canManageFeatureFlagsState } from '@/client-config/states/canManageFeatureFlagsState';
+import { SETTINGS_ADMIN_USER_LOOKUP_WORKSPACE_TABS_ID } from '@/settings/admin-panel/constants/SettingsAdminUserLookupWorkspaceTabsId';
+import { useFeatureFlagsManagement } from '@/settings/admin-panel/hooks/useFeatureFlagsManagement';
+import { useImpersonate } from '@/settings/admin-panel/hooks/useImpersonate';
+import { TextInput } from '@/ui/input/components/TextInput';
+import { TabList } from '@/ui/layout/tab/components/TabList';
+import { useTabList } from '@/ui/layout/tab/hooks/useTabList';
+import { Table } from '@/ui/layout/table/components/Table';
+import { TableCell } from '@/ui/layout/table/components/TableCell';
+import { TableHeader } from '@/ui/layout/table/components/TableHeader';
+import { TableRow } from '@/ui/layout/table/components/TableRow';
+import { DEFAULT_WORKSPACE_LOGO } from '@/ui/navigation/navigation-drawer/constants/DefaultWorkspaceLogo';
+import styled from '@emotion/styled';
+import { isNonEmptyString } from '@sniptt/guards';
+import { useState } from 'react';
+import { useRecoilValue } from 'recoil';
+import { getImageAbsoluteURI, isDefined } from 'twenty-shared';
+import {
+  Button,
+  H1Title,
+  H1TitleFontColor,
+  H2Title,
+  IconSearch,
+  IconUser,
+  Section,
+  Toggle,
+} from 'twenty-ui';
+import { REACT_APP_SERVER_BASE_URL } from '~/config';
+
+const StyledLinkContainer = styled.div`
+  margin-right: ${({ theme }) => theme.spacing(2)};
+  width: 100%;
+`;
+
+const StyledContainer = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+`;
+
+const StyledErrorSection = styled.div`
+  color: ${({ theme }) => theme.font.color.danger};
+  margin-top: ${({ theme }) => theme.spacing(2)};
+`;
+
+const StyledUserInfo = styled.div`
+  margin-bottom: ${({ theme }) => theme.spacing(5)};
+`;
+
+const StyledTable = styled(Table)`
+  margin-top: ${({ theme }) => theme.spacing(3)};
+`;
+
+const StyledTabListContainer = styled.div`
+  align-items: center;
+  border-bottom: ${({ theme }) => `1px solid ${theme.border.color.light}`};
+  box-sizing: border-box;
+  display: flex;
+  gap: ${({ theme }) => theme.spacing(2)};
+`;
+
+const StyledContentContainer = styled.div`
+  flex: 1;
+  width: 100%;
+  padding: ${({ theme }) => theme.spacing(4)} 0;
+`;
+
+export const SettingsAdminGeneral = () => {
+  const [userIdentifier, setUserIdentifier] = useState('');
+  const [userId, setUserId] = useState('');
+
+  const {
+    handleImpersonate,
+    isLoading: isImpersonateLoading,
+    error: impersonateError,
+    canImpersonate,
+  } = useImpersonate();
+
+  const { activeTabId, setActiveTabId } = useTabList(
+    SETTINGS_ADMIN_USER_LOOKUP_WORKSPACE_TABS_ID,
+  );
+
+  const {
+    userLookupResult,
+    handleUserLookup,
+    handleFeatureFlagUpdate,
+    isLoading,
+    error,
+  } = useFeatureFlagsManagement();
```

Comment:
Please breakdown this hook, this is an anti-pattern that creates subscriptions to reactive variables that will inevitably introduce re-renders on the long run. Either break it down in multiple hook for each handler, reactive variable, or directly implement handlers and reactive variables here.